module-utils/utility/Utils.cpp

```diff
@@ -93,10 +93,11 @@ namespace utils
         generator->seed(std::time(nullptr));
         std::uniform_int_distribution<> distribution(0, CHARACTERS.size() - 1);
 
-        std::string random_string(length, '\0');
+        std::string random_string;
+        random_string.reserve(length);
 
-        for (std::size_t i = 0; i < length; ++i) {
-            random_string[i] = CHARACTERS[distribution(*generator)];
+        while (length--) {
+            random_string += CHARACTERS[distribution(*generator)];
```

Comment:
whats exactly changed here? :)

Reply:
didn't string constructor with size in parameter allocate string size if required?
TBC a brief look at the documentation didn't answer my question here but looking at the GCC basic_string header it seems it does. It would be weird if it didn't. 
```
      basic_string(size_type __n, _CharT __c, const _Alloc& __a = _Alloc())
      : _M_dataplus(_M_local_data(), __a)
      { _M_construct(__n, __c); }               <-- here it checks the size and allocates if required.
```

Reply:
Of course, both methods have the same result. I chose `reserved` for reasons I already can't remember. :)

Reply:
Ok, so if it does nothing: then don't change it if it's not needed.